test/core/ticktree/TickTreeTest.t.sol

```diff
@@ -0,0 +1,765 @@
+        }
+
+        int level0Index = level0IndexFromLevel1IndexAndPos(level1Index, level1Pos);
+        Field level0 = mgv.level0(olKey, level0Index);
+        for (uint level0Pos = 0; level0Pos <= MAX_LEVEL0_POSITION; ++level0Pos) {
+          assertEq(
+            isBitSet(level0, level0Pos),
+            isBitSet(tickTree.level0[level0Index], level0Pos),
+            string.concat(
+              "level0 bit mismatch, branch: ",
+              vm.toString(level2Pos),
+              "->",
+              vm.toString(level1Pos),
+              "->",
+              vm.toString(level0Pos)
+            )
+          );
+          if (!isBitSet(level0, level0Pos)) {
+            continue;
+          }
+
+          int leafIndex = leafIndexFromLevel0IndexAndPos(level0Index, level0Pos);
+          Leaf leaf = mgv.leafs(olKey, leafIndex);
+          for (uint leafPos = 0; leafPos <= MAX_LEAF_POSITION; ++leafPos) {
+            {
+              assertEq(
+                leaf.firstOfIndex(leafPos),
+                tickTree.leafs[leafIndex].firstOfIndex(leafPos),
+                string.concat("leaf first mismatch, pos: ", vm.toString(leafPos))
+              );
+              assertEq(
+                leaf.lastOfIndex(leafPos),
+                tickTree.leafs[leafIndex].lastOfIndex(leafPos),
+                string.concat("leaf last mismatch, pos: ", vm.toString(leafPos))
+              );
+            }
+            uint offerId = leaf.firstOfIndex(leafPos);
+            while (offerId != 0) {
+              {
+                MgvStructs.OfferPacked offer = mgv.offers(olKey, offerId);
+                MgvStructs.OfferPacked offerTickTree = tickTree.offers[offerId].offer;
+                assertTrue(
+                  offer.eq(offerTickTree),
+                  string.concat(
+                    "offer mismatch | offerId ",
+                    vm.toString(offerId),
+                    " | MGV ",
+                    toString(offer),
+                    " | tick tree ",
+                    toString(offerTickTree)
+                  )
+                );
+                offerId = offer.next();
+              }
+              {
+                MgvStructs.OfferDetailPacked detail = mgv.offerDetails(olKey, offerId);
+                MgvStructs.OfferDetailPacked detailTickTree = tickTree.offers[offerId].detail;
+                assertTrue(
+                  detail.eq(detailTickTree),
+                  string.concat(
+                    "offer detail mismatch | offerId ",
+                    vm.toString(offerId),
+                    " | MGV ",
+                    toString(detail),
+                    " | tick tree ",
+                    toString(detailTickTree)
+                  )
+                );
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  // # Test tick tree data structure and operations
+
+  struct Offer {
+    MgvStructs.OfferPacked offer;
+    MgvStructs.OfferDetailPacked detail;
+  }
```

Comment:
Why not reuse MgvRoot.OfferData? (maybe not important)

Reply:
Good point. Originally the `Offer` struct was different from `MgvRoot.OfferData` but ended up being identical w/o me realizing 😅

Addressed in 5fd47a7ba1d39d980b20ebf27a2d70752dfe895c